covid_data_getter.py

```diff
@@ -111,45 +112,47 @@ def blankfiller(date,dataset):
 				OpenCSV()
 
 if __name__ == '__main__':
-    import argparse
-    import os
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--csvdir", help="Directory of csv file")
-    args = parser.parse_args()
-    if(args.csvdir):
-        file = os.path.join(args.csvdir, file)
-    # Open the CSV
-    OpenCSV()
-    
-    # Check if there are any missing rows and create if not
-    today = date.today()
-    ftoday = str(today)
-    datesback = 10
-    
-    # Populate any missing dates
-    checkdate = today
-    datesbeckcheck = datesback
-    while datesbeckcheck >0:
-    	OpenCSV()
-    	if dateCheck(str(checkdate)) == 'Date not found':
-    		addRow({'date': checkdate})
-    	checkdate = checkdate - timedelta(days=1)
-    	datesbeckcheck -= 1
-    
-    # Populate missing values 
-    datesbackvalues = datesback
-    checkdate = today
-    while datesbackvalues > 0:
-    	for i in datasources:
-    		OpenCSV()
-    		blankfiller (str(checkdate),i)
-    	checkdate = checkdate - timedelta(days=1)
-    	datesbackvalues -= 1
-    	
-    OpenCSV()
-    
-    localData.sort_values(['date'], 
-                        axis=0,
-                        ascending=[False], 
-                        inplace=True)
-    localData.to_csv(file)
+	import argparse
+	import os
+	parser = argparse.ArgumentParser()
+	parser.add_argument("--csvdir", help="Directory of csv file")
+	args = parser.parse_args()
+	if(args.csvdir):
+		file = os.path.join(args.csvdir, file)
+	# Open the CSV
+	OpenCSV()
+	
+	# Add a blank line to end of CSV if one doesn't exist. (This is needed to add any new rows to the table.)
+	with open(file, 'r') as f:
+		lines = (f.readlines())
+		if '\n' in lines:
+			print: 'new line char found'
+		
+	
+	# Check if there are any missing rows and create if not
+	checkdate = today
+	datesbeckcheck = datesback
+	while datesbeckcheck > 0:
+		OpenCSV()
+		if dateCheck(str(checkdate)) == False:
+			addRow({'date': checkdate})
+		checkdate = checkdate - timedelta(days=1)
+		datesbeckcheck -= 1
```

Comment:
Replace the `while` look with a `for` loop using `range`.  This increases readibility and reduces the number of lines:
```
        for datesback in range(DATESBACK):
                OpenCSV()
                if not dateCheck(str(checkdate)):
                        addRow({'date': checkdate})
                checkdate = checkdate - timedelta(days=1)
```

Reply:
Also note, since you've changed the return type of `dateCheck` to a boolean, you can write `if not dateCheck` rather than checking if it's equal to `False`